pkg/validators/validator_001_default_channel.go

```diff
@@ -0,0 +1,91 @@
+	if success, failureMsg := isListedInChannels(channels, defaultChannel); !success {
+		return false, failureMsg, nil
+	}
+
+	// if success, failureMsg, err := matchesDefaultChannelAnnotations(metabundle.Bundles, defaultChannel); !success {
+	// 	return false, failureMsg, err
+	// }
+	return true, "", nil
+}
+
+func isPartOfEnum(defaultChannel string) (bool, string) {
+	enum := map[string]struct{}{
+		"alpha":  {},
+		"beta":   {},
+		"stable": {},
+		"edge":   {},
+		"rc":     {},
+	}
+	if _, ok := enum[defaultChannel]; !ok {
+		return false, fmt.Sprintf("The defaultChannel '%v' is not part of the accepted values: alpha, beta, stable, edge or rc.", defaultChannel)
+	}
+	return true, ""
+}
+
+// TODO - deprecate this when we remove legacy builds
+func isListedInChannels(channels *[]v1alpha1.Channel, defaultChannel string) (bool, string) {
+	// as the Channels field is deprecated, it can be omitted
+	if channels == nil {
+		return true, ""
+	}
+	var channelNames []string
+	for _, channel := range *channels {
+		if channel.Name == defaultChannel {
+			return true, ""
+		}
+		channelNames = append(channelNames, channel.Name)
+	}
+	return false, fmt.Sprintf("The defaultChannel '%v' is not part of the listed channelNames: %v.", defaultChannel, channelNames)
+}
+
+// TODO - (sblaisdo) enable when extraction format is bundles instead of packageManifest
+// if the annotation is not present, make sure the defaultChannel is alpha
+// func matchesDefaultChannelAnnotations(bundles []registry.Bundle, defaultChannel string) (bool, string, error) {
+// 	for _, bundle := range bundles {
+// 		allChannels := []string{bundle.Annotations.Channels, "dontexist"}
+// 		var defaultBundleChannel string
+
+// 		if len(allChannels) == 1 {
+// 			// if a single channel is listed, should match defaultChannel
+// 			defaultBundleChannel = allChannels[0]
+
+// 		} else {
+// 			// if multiple channels are listed, need to specify the optional
+// 			// operators.operatorframework.io.bundle.channel.default.v1
+// 			defaultBundleChannel = bundle.Annotations.DefaultChannelName
+// 		}
+// 		if defaultChannel != defaultBundleChannel {
+// 			version, err := bundle.Version()
+// 			if err != nil {
+// 				return false, "", fmt.Errorf("Could not read bundle version, got %v.", err)
+// 			}
+// 			msgs := []string{fmt.Sprintf("Missing or invalid channel annotation for bundle '%v:%v'", bundle.Name, version)}
+// 			msgs = append(msgs, fmt.Sprintf("DefaultChannel '%v' does not match channel from bundle annotations '%v'.", defaultChannel, defaultBundleChannel))
+// 			if len(allChannels) == 1 {
+// 				msgs = append(msgs, fmt.Sprintf("Please set the 'operators.operatorframework.io.bundle.channels.v1' annotation to '%v'.", defaultChannel))
+// 			} else {
+// 				msgs = append(msgs, fmt.Sprintf("Please set the operators.operatorframework.io.bundle.channel.default.v1 annotation to '%v'.", defaultChannel))
+// 			}
+// 			return false, strings.Join(msgs, "\n"), nil
```

Comment:
```suggestion
// 			return false, strings.Join(msgs, ": "), nil
```
Minor nit :)
(coz we're (and usually, as far as I have noticed) delimiting error/failure strings with ": ")

Reply:
I don't have a strong opinion on this. We just have to keep in mind those error msgs are printed both on stdout (CLI validation) and eventually in the validation-webhook-server response payload. Fixed!